sycl/include/CL/sycl/handler.hpp

```diff
@@ -720,14 +720,15 @@ class __SYCL_EXPORT handler {
   // NOTE: the name of this function - "kernel_single_task" - is used by the
   // Front End to determine kernel invocation kind.
   template <typename KernelName, typename KernelType>
-  __attribute__((sycl_kernel)) void kernel_single_task(KernelType KernelFunc) {
+  __attribute__((sycl_kernel)) void
+  kernel_single_task(const KernelType KernelFunc) {
```

Comment:
Minor comment: kernel_single_task() is called only from single_task(), right?
If so, then 'const' in this line adds nothing because KernelType is already a const type (KernelFunc arg is 'const KernelType' type in single_task()). 
Same for kernel_parallel_for() and kernel_parallel_for_work_group().

This additional 'const' is harmless though.

Reply:
@v-klochkov  - what about the reductions?  They have KernelType definitions as well, and I presume the same const reference constraint will apply to them? But if they are based on these other constructs, that might be enough.  I haven't looked to know.

Reply:
> @v-klochkov - what about the reductions? They have KernelType definitions as well, and I presume the same const reference constraint will apply to them? But if they are based on these other constructs, that might be enough. I haven't looked to know.

@Pennycook wrote the feature proposal for reduction and he already answered below that parallel_for() accepting reduction should be in sync with older parallel_for not accepting reductions. Probably, the proposal for reduction may need a minor fix to follow this changes.